packages/dds/tree/src/core/edit-manager/editManager.ts

```diff
@@ -309,6 +316,26 @@ export class EditManager<
 		return this.changeFamily.intoDelta(change);
 	}
 
+	/**
+	 * Fast-forwards the local branch to a new head commit that is ahead of the current local branch head.
+	 * @param newHead - the new head of the local branch. This must be a descendant of the current local branch head.
+	 * @returns the composition of changes from the previous head to the new head of the local branch
+	 */
+	public fastForwardLocalBranch(newHead: GraphCommit<TChangeset>): TChangeset {
```

Comment:
The name and description of this function lead to me to expect that `this.localBranch` would be set to `newHead` by the time this function returns.

Reply:
Yea, that is confusing, you are right. It also made me realize that we are currently squashing the checkout as it is merged into the root, which we don't want to do necessarily. I'll refactor this part.

Reply:
I was able to remove this function altogether at the small cost of having to import GraphCommits into SharedTreeCore.